charts/datadog/templates/daemonset.yaml

```diff
@@ -99,8 +99,8 @@ spec:
       imagePullSecrets:
 {{ toYaml .Values.agents.image.pullSecrets | indent 8 }}
       {{- end }}
-      {{- if .Values.agents.priorityClassName }}
-      priorityClassName: {{ .Values.agents.priorityClassName }}
+      {{- if or .Values.providers.gke.autopilot .Values.agents.priorityClassName }}
```

Comment:
IMO we can create the PriorityClass all the time.

Reply:
my fear if we create the PC all the time is that it's not necessary in most cases and I can foresee users coming back to us having not seen that it's now a default and whether they have one already or have a lifecycle that depends on it for their applications, this could cause some inconvenience.
As it is an advanced feature, I believe we should keep it optional.

Reply:
Yes fare enough.
in that case, we can add a new parameter `agent.createPriorityClass: false`